draft-ietf-dprive-dnsoquic.md

```diff
@@ -663,29 +702,61 @@ attacker might replay it at chosen times, several times.
 
 The recommendation for TLS 1.3 {{?RFC8446}} is that the capability to use 0-RTT
 data should be turned off by default, and only enabled if the user clearly
-understands the associated risks.
-
-QUESTION: Should 0-RTT only be used with Opportunistic profiles (i.e.
-disabled by default for Strict only)?
+understands the associated risks. In our case, allowing 0-RTT data
+provides significant performance gains, and we are concerned that a
+recommendation to not use it would simply be ignored. Instead, we provide
+a set of practical recommendations in {{connection-resume-and-0-rtt}}
+and {{using-0-rtt-and-resumption}}.
+
+The prevention on allowing replayable transactions in 0-RTT data
+expressed in {{connection-resume-and-0-rtt}} blocks the most obvious
+risks of replay attacks, as it only allows for transactions that will
+not change the long term state of the server. 
+
+Attacks trying to assess the state of the cache are more powerful if
+the attacker can choose the time at which the 0-RTT data will be replayed.
+We believe that the freshness tests recommended in {{connection-resume-and-0-rtt}}
+significantly reduce the time range in which 0-RTT data can be
+replayed, and thus significantly reduce the potential of such
+attacks. The maximum delay parameter stated in {{connection-resume-and-0-rtt}}
+is 30 seconds. We believe this is consistent with commonly used values
+of the cached records TTL, and thus sufficiently small to impede
+most replay attacks.
```

Comment:
Now we get to some rationale for the 30s value.  If the goal is to ensure that the replay window is shorter than the minimum cache retention time, that sounds reasonable.  In that case, please express the requirement in those terms, rather than pick a value seemingly at random.

That is, I think that you are trying to say:

> When deciding whether to accept 0-RTT, servers need to allow for some variation in the calculated ticket age; see {{Section 8.3 of RFC8446}}. This allows for both variation in network round trip time plus drift in the clocks of client and server relative to each other.  In a caching resolver, the length of the window over which 0-RTT is accepted SHOULD be less than the minimum time that a record will be cached.  (Then you need to reference the text on the attack, which needs to be elsewhere, I think.)

All that said, I'm not sure that this helps much.  The goal seems to be to stop an attacker from using replays of the same query to observe timing and then infer the precise time that the record was evicted, and from that tell which record was being queried.  However, that's something that is possible no matter how wide the window is.  The tolerance only narrows the time over which replays might be allowed.  (FWIW, NSS uses a node-specific bloom filter for anti-replay, which means that you would have to hit different nodes for each replay, but that also means that you are likely hitting a different cache.)

The only advice that really helps here is to reduce the tolerance as much as you can.  Which is advice that TLS already provides.